src/tools/interop/idt/capture/pcap/pcap.py

```diff
@@ -0,0 +1,62 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import os
+import time
+
+from capture.file_utils import create_standard_log_name
+from capture.shell_utils import BashRunner
+
+
+class PacketCaptureRunner:
+
+    def __init__(self, artifact_dir: str, interface: str) -> None:
+
+        self.artifact_dir = artifact_dir
+        self.output_path = str(
+            os.path.join(
+                self.artifact_dir,
+                create_standard_log_name(
+                    "pcap",
+                    "cap")))
+        self.start_delay_seconds = 3
+        self.interface = interface
+        self.pcap_proc: BashRunner | None = None
+        self.pcap_command = f"tcpdump -i {self.interface} -n -w {self.output_path}"
+
+    def start_pcap(self) -> None:
+        if self.pcap_proc is None or not self.pcap_proc.command_is_running():
```

Comment:
You might want to cleanup the calling pattern so that there is a clear entrypoint. It seems sloppy that it's internally trying to handle clients that are using a bad call pattern. I think I would probably raise an Illegal State Exception if a client calls start when it's already running.

Reply:
Thanks I'll need to take a second look at this. Android followed the same pattern, and it is called multiple times without issue during capture.

Reply:
Thanks, have taken care of this now!